internal/pkg/api/handleEnroll.go

```diff
@@ -96,12 +95,27 @@ func (et *EnrollerT) handleEnroll(zlog zerolog.Logger, w http.ResponseWriter, r
 	return writeResponse(zlog, w, resp, ts)
 }
 
-func (et *EnrollerT) processRequest(zlog zerolog.Logger, w http.ResponseWriter, r *http.Request, rb *rollback.Rollback, enrollmentAPIKeyID, ver string) (*EnrollResponse, error) {
+func (et *EnrollerT) processRequest(zlog zerolog.Logger, w http.ResponseWriter, r *http.Request, rb *rollback.Rollback, enrollmentAPIKey *apikey.APIKey, ver string) (*EnrollResponse, error) {
+	var enrollAPI *model.EnrollmentAPIKey
+	if et.cfg.StaticPolicyTokens.Enabled {
```

Comment:
I think we should try to keep our authentication handling in `handleEnroll` (currently line 76), we may need to move some existing logic out of `processRequest` to do so

Reply:
This is done to prevent the call to elastic to fetch the token since it does not really exists there.